### ISSUE TITLE
services -u not working correctly

### ISSUE THREAD

**Issue by whoot** [rapid7/metasploit-framework#10867]
## Steps to reproduce

1. import an nmap scan with open and closed ports
2. request only open ports with 'service -u'

## Expected behavior

Show only services which are actually 'open'.

## Current behavior

All ports - 'open' AND 'closed' - are shown.
![services](https://user-images.githubusercontent.com/8066322/47496693-e923ec00-d857-11e8-8571-a9505533dc6f.png)

### Metasploit version

Framework: 5.0.0-dev-6920470f99
Console  : 5.0.0-dev-6920470f99

### I installed Metasploit with:
- git (community edition)

### OS

Linux kali 4.18.0-kali2-amd64 #1 SMP Debian 4.18.10-2kali1 (2018-10-09) x86_64 GNU/Linux

**Comment by bcoles** [comment]
Reproduced this issue on latest Metasploit from GitHub

**Comment by bcoles** [comment]
I've submitted a fix for this in #10945; although it's worth noting that I didn't bother looking into what caused the `--up` flag to stop working.

**Comment by bcoles** [comment]
To reproduce this issue:

```
# nmap -d -d -F somehost.local -oX /home/user/hackyhackhack
# # maybe also try a UDP scan:
# nmap -sU -d -d -F somehost.local -oX /home/user/hackyhackhack
```

```
msf > db_import /home/user/hackyhackhack
msf > services
msf > services --up
```
